northstar/src/npk/npk.rs

```diff
@@ -635,58 +644,115 @@ fn sign_npk(key: &Path, fsimg: &Path, manifest: &Manifest) -> Result<String, Err
+            .skip(1)
+            .try_fold(PathBuf::from("/"), |mut p, dir| {
+                p.push(dir);
+                writeln!(file, "{} d {} {} {}", p.display(), mode, uid, gid).map_err(|e| {
+                    Error::Io {
+                        context: format!("Failed to write entry {} to temp file", p.display()),
+                        error: e,
+                    }
+                })?;
+                Ok(p)
+            })
+            .map(drop)
     }
 
+    // Create mountpoints as pseudofiles/dirs
     for (target, mount) in &manifest.mounts {
-        let mode = match mount {
+        match mount {
             Mount::Bind(Bind { options: flags, .. }) => {
-                if flags.contains(&MountOption::Rw) {
-                    777
+                let mode = if flags.contains(&MountOption::Rw) {
+                    755
                 } else {
                     555
-                }
+                };
+                add_directory(file, target, mode, uid, gid)?;
             }
-            Mount::Persist => 777,
-            Mount::Resource { .. } => 555,
-            Mount::Tmpfs { .. } => 777,
-            // /dev is default
-            Mount::Dev { .. } => continue,
-        };
+            Mount::Persist => add_directory(file, target, 755, uid, gid)?,
+            Mount::Proc => add_directory(file, target, 444, uid, gid)?,
+            Mount::Resource { .. } => add_directory(file, target, 555, uid, gid)?,
+            Mount::Tmpfs { .. } => add_directory(file, target, 755, uid, gid)?,
+            Mount::Dev => {
+                // Create a minimal set of chardevs:
+                // └─ dev
+                //     ├── fd -> /proc/self/fd
+                //     ├── full
+                //     ├── null
+                //     ├── random
+                //     ├── stderr -> /proc/self/fd/2
+                //     ├── stdin -> /proc/self/fd/0
+                //     ├── stdout -> /proc/self/fd/1
+                //     ├── tty
+                //     ├── urandom
+                //     └── zero
+
+                // Create /dev pseudo dir. This is needed in order to create pseudo chardev file in /dev
+                add_directory(file, target, 755, uid, gid)?;
+
+                // Create chardevs
+                for (dev, major, minor) in &[
+                    ("full", 1, 7),
+                    ("null", 1, 3),
+                    ("random", 1, 8),
+                    ("tty", 5, 0),
+                    ("urandom", 1, 9),
+                    ("zero", 1, 5),
+                ] {
+                    writeln!(
+                        file,
+                        "{} c {} {} {} {} {}",
+                        target.join(dev).display(),
+                        666,
+                        uid,
+                        gid,
+                        major,
+                        minor
+                    )
+                    .map_err(|e| Error::io("Failed to write pseudofile", e))?;
```

Comment:
nit: maybe move this bolck to a `add_chardevs` together with `add_directory` and have them documented/tested